[pzaiming] iP

pzaiming opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

pzaiming committed:
Level-1

pzaiming committed:
Level-1

pzaiming committed:
Level-1

pzaiming committed:
Level-2

pzaiming committed:
Level-3

pzaiming committed:
Level-4

pzaiming committed:
Level-5

pzaiming committed:
Level-5

pzaiming committed:
Level-6

pzaiming committed:
A-TextUiTesting

pzaiming committed:
A-TextUiTesting

pzaiming committed:
Level-5

pzaiming committed:
Level-6

pzaiming committed:
A-Enums

pzaiming committed:
branch-Level-7

pzaiming committed:
Revert "branch-Level-7"

This reverts commit fe961fed5e4ddbd30aeb3bff0cf794da224e77c3.

pzaiming committed:
Level-7

pzaiming committed:
Level-7

pzaiming committed:
Level-8: Printing the dates in the format such MMM dd yyyy

pzaiming committed:
Level-8: Printing the dates in the format such MMM dd yyyy

pzaiming committed:
Level-8: Updated the date and time to allow deadline to accept date format as such: deadline {task} /by {yyyy-mm-dd hhmm}

pzaiming committed:
Merge branch 'branch-Level-7'

pzaiming committed:
Merge branch 'branch-Level-8'

pzaiming committed:
Merge remote-tracking branch 'origin/add-gradle-support'

pzaiming committed:
A-Gradle

pzaiming committed:
A-MoreOOP: Change the code to more OOP orientated by refactoring the code to extract out closely related code as classes.

pzaiming committed:
A-Packages: Divide classes into packages

pzaiming committed:
A-JUnit: Add JUnit tests to test the behavior for Todo, Deadline, Event Tasks.

pzaiming committed:
A-JavaDoc: Add header comments to all non-private classes/methods, and non-trivial private methods.

pzaiming committed:
A-JavaDoc: Add header comments to all non-private classes/methods, and non-trivial private methods.

pzaiming committed:
A-MoreOOP: Rectified the bugs in the code.

pzaiming committed:
Level-8: Updated code

pzaiming committed:
A-MoreOOP: Made changes to the code as there were bugs.

pzaiming committed:
A-Packages: Updated to fix bugs in the code.

pzaiming committed:
A-JUnit: Updated to fix bugs in the code.

pzaiming committed:
A-JavaDoc: Updated to the documentation of the code.

pzaiming committed:
A-JavaDoc: Updated to the documentation of the code.

pzaiming committed:
A-CodingStandard: Tweak the code to comply with a coding standard.

pzaiming committed:
Level-9: Give users a way to find a task by searching for a keyword.

pzaiming committed:
Level-9: Give users a way to find a task by searching for a keyword.

pzaiming committed:
Level-9: Give users a way to find a task by searching for a keyword.

pzaiming committed:
Added Level-9 features

pzaiming committed:
Merge remote-tracking branch 'origin/branch-Level-8'

pzaiming committed:
Merge remote-tracking branch 'origin/branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/exception/DukeException.java
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/storage/Storage.java
#	src/main/java/duke/task/Task.java

pzaiming committed:
Merge remote-tracking branch 'origin/branch-A-CodingStandard'

pzaiming committed:
Merge remote-tracking branch 'origin/branch-Level-9'

pzaiming committed:
Resolved merging conflicts.

pzaiming committed:
Credited resued codes.

bernicetoh commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+            tasks = new Storage();
+        }
+    }
+
```
Perhaps you can remove the unnecessary line breaks?

bernicetoh commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,35 @@
+ */
+public class FindCommand extends Command {
+    private String keyword;
+
```
Unnecessary line break here as well.

bernicetoh reviewed (commented):
Overall, I think your code generally follows the coding standard except for a few instances where you could remove the unnecessary line breaks between class members.

gyulong1 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+            } catch (DukeException e) {
+                System.out.println(e);
+            } catch (IllegalArgumentException e) {
+                ui.showLoadingError("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
```
Perhaps you could consider including this string in the UI class instead.

gyulong1 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,71 @@
+    public static boolean isValidCommand(String cmd) throws DukeException {
+        String cmdtype = cmd.split(" ")[0];
+        // check that duke.command list must not have any more description
+        if (cmdtype.equals("list") && cmdtype.length() != cmd.length()) {
```
Can consider using switch statements to improve readability.

gyulong1 commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,35 @@
+
+    public boolean execute(Storage tl, Ui ui, Storage storage) {
+        Task t = new Deadline(this.activity, this.date, this.time);
+        tl.addTask(t);
```
Should t be named as task instead?

gyulong1 commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,24 @@
+    }
+
+    public boolean execute(Storage tl, Ui ui, Storage storage) {
+        Task t6 = tl.removeTask(this.id - 1);
```
t6 could be renamed to something more relevant

gyulong1 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,59 @@
+                return new FindCommand(cmd);
+            default:
+                throw new IllegalArgumentException();
+        }
```
Great job, code is very readable even though there are many cases.

gyulong1 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,29 @@
+ */
+public class Deadline extends Task {
+
+    private LocalDate bydate;
```
variables should be in camelcase

gyulong1 reviewed (commented):
Great job! Code is mostly readable and neat, OOP is also done naturally.

pzaiming committed:
Made changes from the Peer Review.

pzaiming committed:
A-MoreOOP: Added TaskList to make it more OOP.

rockman007372 commented on data/duke.txt:
```diff
@@ -1,5 +1,4 @@
-3. [T][ ] borrow book
-4. [D][ ] return book (by: Dec 2 2023 01:50 PM)
-5. [E][ ] project meeting (from: Mon 2pm to: 4pm)
+3. [D][ ] return book (by: Dec 2 2023 01:50 PM)
```
If I am not mistaken, I don't think we should push the /data/ directory into the remote repo. This is a comment I received from another reviewer so I could be mistaken.

rockman007372 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -1,3 +1,7 @@
+//@@author pzaiming-reused
```
It's great practice that you credited the code owner! But I think the instruction mentioned that you have to enclose the reused code block with the `//@@author` tags, meaning **both** the start and the end of the code block have to be marked!

```Java
//@@author {yourGithubUsername}-reused
//{Info about the source...}

{Reused code (possibly with minor modifications) here ...}

//@@author
```

rockman007372 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -100,6 +100,9 @@ public ArrayList<Task> getTasks() {
     public ArrayList<Task> load() throws DukeException {
+        //@@author pzaiming-reused
+        //Reused from https://github.com/RyanQiu1
+        // with minor modifications
```
I cannot comment on the method `Storage.updateStorage()` because it was not modified in this commit. However, I think the Javadoc of this method does not abide by coding convention. 

I think `This method will update the textfile based on the current commands inputed by the user.` should be changed to ` Updates the textfile based on the current commands inputed by the user` to align with convention.

Example of a Javadoc comment:

```Java
/**
 * Returns lateral location of the specified position.
 * If the position is unset, NaN is returned.
 *
 * @param x X coordinate of position.
 * @param y Y coordinate of position.
 * @param zone Zone of position.
 * @return Lateral location.
 * @throws IllegalArgumentException  If zone is <= 0.
 */
public double computeLocation(double x, double y, int zone)
        throws IllegalArgumentException {
    //...
}
```

rockman007372 commented on src/test/java/duke/task/TodoTest.java:
```diff
@@ -1,3 +1,6 @@
+//@@author pzaiming-reused
+//Reused from https://github.com/RyanQiu1
+// with minor modifications
```
Again I cannot leave comment on the TodoTest class because it was not modified. However, I think you can change the name of the method `dummyTest()` to something of the form `whatIsBeingTested_descriptionOfTestInputs_expectedOutcome` for better clarity.

rockman007372 reviewed (commented):
Neat clean code with good use of OOP overall. My suggestions are to check the Javadoc comment for public methods to make sure it refers to the method in third person (in accordance with Java Coding Convention).  Also more test cases would be beneficial for future debugging.

pzaiming committed:
A-CheckStyle: Use checkStyle to detect coding style violations.

pzaiming committed:
Level-10: added basic GUI interface.

pzaiming committed:
Level-10: added basic GUI interface.

pzaiming committed:
Add assert features

Assert statements can be used to document important assumptions that should hold at various points in the code.

pzaiming committed:
Improve Code Quality

Critically examines the code and refactor to improve the code quality where necessary.

Making changes to different parts of the code such as by removing magic strings.

To ensure the code have better readability.

pzaiming committed:
Merge pull request #2 from pzaiming/branch-A-Assertions

Add assert features

pzaiming committed:
Merge pull request #3 from pzaiming/branch-A-CodeQuality

Improve Code Quality

pzaiming committed:
Add Extensions C-DetectDuplicates

Add the ability to recognize and deal with duplicate tasks.

pzaiming committed:
Merge pull request #4 from pzaiming/branch-C-DetectDuplicates

Add Extensions C-DetectDuplicates

pzaiming committed:
A-CI

Use GitHub Actions to set up Continuous Integration (CI).

pzaiming committed:
A-CI

Use GitHub Actions to set up Continuous Integration (CI).

pzaiming committed:
Use GitHub Actions to set up Continuous Integration (CI)

pzaiming committed:
Use GitHub Actions to set up Continuous Integration (CI)

pzaiming committed:
Use GitHub Actions to set up Continuous Integration (CI)

pzaiming committed:
Use GitHub Actions to set up Continuous Integration (CI)

pzaiming committed:
Improve the GUI.

pzaiming committed:
A-Personality: Give a unique personality.

pzaiming committed:
A-UserGuide: Add a User Guide and also fix some bugs in the code.

pzaiming committed:
A-UserGuide: Add a User Guide and also fix some bugs in the code.

pzaiming committed:
A-UserGuide: Add a User Guide and also fix some bugs in the code.

pzaiming committed:
A-UserGuide: Add a User Guide and also fix some bugs in the code.

pzaiming committed:
Update deadline task to prevent older datetime as input from the user being added to the list.

pzaiming committed:
Update deadline task to prevent older datetime as input from the user being added to the list.

pzaiming committed:
Update the docs of some methods.
